idem_provider_azurerm/states/azurerm/keyvault/key.py

```diff
@@ -0,0 +1,314 @@
+                client_id: ABCDEFAB-1234-ABCD-1234-ABCDEFABCDEF
+                secret: XXXXXXXXXXXXXXXXXXXXXXXX
+                cloud_environment: AZURE_PUBLIC_CLOUD
+
+'''
+# Python libs
+from __future__ import absolute_import
+import logging
+
+log = logging.getLogger(__name__)
+
+TREQ = {
+    'present': {
+        'require': [
+            'states.azurerm.keyvault.vault.present',
+        ]
+    }
+}
+
+
+async def present(hub, ctx, name, key_type, vault_url, key_ops=None, enabled=None, expires_on=None, not_before=None,
+                  tags=None, connection_auth=None, **kwargs):
+    '''
+    .. versionadded:: VERSION
+
+    Ensure the specified key exists within the given key vault. Requires keys/create permission. Key properties can be
+        specified as keyword arguments.
+
+    :param name: The name of the new key. Key names can only contain alphanumeric characters and dashes.
+
+    :param key_type: The type of key to create. Possible values include: 'ec', 'ec_hsm', 'oct', 'rsa', 'rsa_hsm'.
+
+    :param vault_url: The URL of the vault that the client will access.
+
+    :param key_ops: A list of permitted key operations. Possible values include: 'decrypt', 'encrypt', 'sign',
+        'unwrap_key', 'verify', 'wrap_key'.
+
+    :param enabled: Whether the key is enabled for use.
+
+    :param expires_on: When the key will expire, in UTC. This parameter must be a string representation of a Datetime
+        object in ISO-8601 format.
+
+    :param not_before: The time before which the key can not be used, in UTC. This parameter must be a string
+        representation of a Datetime object in ISO-8601 format.
+
+    :param tags: A dictionary of strings can be passed as tag metadata to the key.
+
+    :param connection_auth: A dict with subscription and authentication parameters to be used in connecting to the
+        Azure Resource Manager API.
+
+    Example usage:
+
+    .. code-block:: yaml
+
+        Ensure key exists:
+            azurerm.keyvault.key.present:
+                - name: my_key
+                - key_type: my_type
+                - vault_url: my_vault
+                - tags:
+                    contact_name: Elmer Fudd Gantry
+                - connection_auth: {{ profile }}
+
+    '''
+    ret = {
+        'name': name,
+        'result': False,
+        'comment': '',
+        'changes': {}
+    }
+
+    if not isinstance(connection_auth, dict):
+        ret['comment'] = 'Connection information must be specified via connection_auth dictionary!'
+        return ret
+
+    key = await hub.exec.azurerm.keyvault.key.get_key(
+        name=name,
+        vault_url=vault_url,
+        azurearm_log_level='info',
+        **connection_auth
+    )
+
+    if key_type != 'oct':
+        key_type = key_type.upper().replace('_', '-')
+
+    if 'error' not in key:
+        if tags:
+            tag_changes = await hub.exec.utils.dictdiffer.deep_diff(key.get('properties').get('tags', {}), tags or {})
```

Comment:
`get('properties')` needs to default to an empty dict or else you get:
`TypeError: 'NoneType' object is not iterable`